src/aggregates/vector_selector.rs

```diff
@@ -1,3 +1,123 @@
+//! # Vector Selector
+//! The `vector_selector` aggregate implements Prometheus' `VectorSelector` processing.
+//!
+//! ## Instant Vector
+//! A Prometheus `Instant Vector` is a set of time series, each with a single data point. The single
+//! data point of each series is determined by looking back over a window of time (`lookback`) from
+//! the reference point `t` and retrieving the most recent sample.
+//!
+//! As an example, we see that given an evaluation time `t`, we look back on the timeseries `ts`
+//! over a window of size `lookback` for the most recent sample, in this case `e`.
+//!
+//! ```text
+//!                          t
+//!                          |
+//!  ts:  a b   c    d   e     f  g
+//!                      ^
+//! out:                     e
+//!          |---lookback----|
+//! ```
+//!
+//! ## Range Vector
+//!
+//! This can be applied over a time range to determine a `Range Vector`, which can be thought of as
+//! the `Instant Vector` for multiple reference points `(t_1, t_2, ..., t_n)`.
+//!
+//! The `Range Vector` is determined by a `start_time`, an `end_time`, a `bucket_width`, and
+//! `lookback`. These dictate the times `t_i` at which we look back over the time series to find
```

Comment:
Hmm for range vectors there is no lookback. You always consider the entire bucket. In other words the lookback=bucket_width

Reply:
Slowly my mental model of this is starting to take shape. Thanks!